### PR TITLE
Fixed RD-10971

### PR DESCRIPTION
**Pull request by bgaidioz**
The crash was hit because we use `token.toString` when the `token` can be `null`. Instead I used `String.valueOf(token)`.

Also added tests for bugs hit while debugging that.

### CODE REVIEW

File: snapi-compiler/src/test/scala/com/rawlabs/snapi/compiler/tests/regressions/RD10971Test.scala [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    // TODO: RD-10971: the record is indeed parsed as a failed record, but we skip the following one (Jane)
+    it should evaluateTo(
+      """[{name: "Benjamin", birthYear: 1978},
+        |Error.Build("'birthYear': not found"),
+        |{name: "Jane", birthYear: 200}]""".stripMargin
+    )
+  }
+
+  // Fails because of RD-10971
+  ignore(
```
Comment (bgaidioz) [inline snapi-compiler/src/test/scala/com/rawlabs/snapi/compiler/tests/regressions/RD10971Test.scala:57]: Add a test that mimics the actual RD-1971 bug (parsing an empty string as a collection of records).

And document the fix that was tried.

**Review by torcato — APPROVED** [review]
LGTM
